libpoolprotocols/stratum/EthStratumClient.cpp

```diff
@@ -173,13 +173,21 @@ void EthStratumClient::disconnect()
 	m_responsetimer.cancel();
 	m_response_pending = false;
 
-	if (m_secureMode != StratumSecure::NONE) {
-		boost::system::error_code sec;
-		m_securesocket->shutdown(sec);
-	}
+	try {
+		if (m_secureMode != StratumSecure::NONE) {
+			boost::system::error_code sec;
+			m_securesocket->shutdown(sec);
+		}
 
-	m_io_service.stop();
-	m_socket->close();
+		m_io_service.stop();
+#ifdef _WIN32
```

Comment:
This is not necessary. The socket close is ok in Linux if you move the m_io_service.stop() after it. Makes sense!

Reply:
Doesn't help for SSL though.